### PR TITLE
✨ feat(form): horizontal layout for label and controls

### PR DESCRIPTION
**Pull request by mladenplaninicic**

### CODE REVIEW

**Comment by incubator-tekton** [comment]
:rocket: Welcome to Baloise ChatOps based GitOps Bot :tada:
                                
    - If you want to enable preview environments for this PullRequest, comment with `/preview`. 
    - If you want to disable the preview environment for this PR, comment with `/undeploy`. 
    - If you need help or want to see the docs, comment with `/help`. 
                                
    Make sure to refresh this page after submitting a command.
    Preview environments can be redeployed automatically if you enable the webhook for updates to the source branch. :new:
    
    Have fun building awesome apps! :hammer:  :whale2:  :truck:  :ship:

File: packages/components/src/components/bal-field/bal-field.vars.sass [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @prop --bal-field-label-horizontal-width: Width of the label
+ * @prop --bal-field-label-column-gap: Gap between the label and the input
+ */
+
+:root
+  //
+  // label
+  --bal-field-label-horizontal-width: 33%
```
Comment (hirsch88) [inline packages/components/src/components/bal-field/bal-field.vars.sass]: use `33.3333`

File: test/cypress/snapshots/base/visual/bal-field.visual.cy.ts/horizontal-alignment-desktop.png
Comment (hirsch88) [inline test/cypress/snapshots/base/visual/bal-field.visual.cy.ts/horizontal-alignment-desktop.png:1]: - [x] add visual test for long labels and message

**Review by hirsch88 — CHANGES_REQUESTED** [review]

File: packages/components/src/components.d.ts [hunk 2]
```diff
@@ -966,6 +966,10 @@ export namespace Components {
           * If `true`, the element is not mutable, focusable, or even submitted with the form. The user can neither edit nor focus on the control, nor its form control descendants.
          */
         "disabled"?: boolean;
+        /**
+          * If true, label and input are aligned horizontally within the field component, with the message positioned in a new line below.
+         */
+        "horizontalAlignment"?: boolean;
```
Comment (hirsch88) [inline packages/components/src/components.d.ts]: - [x] rename to horizontal

File: packages/components/src/components/bal-field/bal-field.sass [hunk 3]
```diff
@@ -43,3 +44,36 @@
...
+    margin-left: calc(var(--bal-field-label-horizontal-width) + var(--bal-field-label-column-gap))
+
+  .bal-field-hint
+    margin-left: 0
+
+    .bal-hint
+      top: 0
+
+  .bal-field-hint ~ .bal-field-message 
+    margin-left: calc(var(--bal-field-label-horizontal-width) + (var(--bal-field-label-column-gap) * 2) + 18px)
```
Comment (hirsch88) [inline packages/components/src/components/bal-field/bal-field.sass]: what is 18px?

File: .changeset/slow-parrots-argue.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (hirsch88) [inline .changeset/slow-parrots-argue.md:1]: - [x] add the documentation to storybook

**Review by hirsch88 — CHANGES_REQUESTED** [review]

**Review by hirsch88 — APPROVED** [review]

**Pull request merged**